apollo-federation/src/sources/connect/validation/schema.rs

```diff
@@ -327,14 +345,145 @@ fn advanced_validations(schema: &SchemaInfo, subgraph_name: &str) -> Vec<Message
     entity_checker.check_for_missing_entity_connectors(schema)
 }
 
+fn compute_input_trie(connector: &Connector) -> SelectionTrie {
+    let mut merged = SelectionTrie::new();
+    for param in connector.variable_references() {
+        merged.extend(&param.selection);
+    }
+    merged
+}
```

Comment:
```suggestion
fn compute_input_trie(connector: &Connector) -> SelectionTrie {
    let mut merged = SelectionTrie::new();
    for param in connector.variable_references() {
        merged.add_str(param.namespace.as_str()).extend(&param.selection);
    }
    merged
}
```
I may be missing how this is accounted for elsewhere, but I think you want to preserve the name of each variable at the top level of the trie (note `param.selection` only represents the selection set for that single variable, so it does not already include the variable name).

Reply:
I think `param.namespace.as_str()` is not valid

Reply:
Sorry, I wrote the code above without testing it locally first. I hope the intent was clear enough, and that you feel empowered to remove `#[cfg(test)]` (or find a similar method) to get it working, according to your judgement.

Reply:
We only actually care about the `$batch` namespace here so it is only necessary to compute the trie for that namespace. I will update to correct.